src/arbcall.jl

```diff
@@ -100,24 +99,47 @@ function jlcode(af::Arbfunction, jl_fname=jlfname(af))
     if !isnothing(k)
         @assert jl_types[k] == arb_rnd
         r = :rnd
-        push!(kwargs, Expr(:kw, :($r::Union{arb_rnd, RoundingMode}), :(RoundNearest)))
+        push!(jl_kwargs, Expr(:kw, :($r::Union{arb_rnd, RoundingMode}), :(RoundNearest)))
         deleteat!(arg_names, k)
         deleteat!(jl_types, k)
     end
 
     jl_args = [:($a::$T) for (a, T) in zip(arg_names, jl_types)]
 
-    res = first(arg_names)
+    :($(jlfname(af))($(jl_args...); $(jl_kwargs...))::$(returnT))
+end
+
+function arbsignature(af::Arbfunction)
+    jltoctype = Dict(value => key for (key, value) in Ctypes)
 
-    return :(
-        function $jl_fname($(jl_args...); $(kwargs...))
-            ccall(Arblib.@libarb($(arbfname(af))),
-            $returnT,
-            $(Expr(:tuple, c_types...)),
-            $(Symbol.(name.(args))...))
-            return $res
-        end
-        )
+    creturnT = jltoctype[returntype(af)]
+    args = arguments(af)
+
+    arg_consts = isconst.(args)
+    arg_ctypes = [jltoctype[jltype(arg)] for arg in args]
+    arg_names = name.(args)
+
+
+    c_args = join([ifelse(isconst, "const ", "")*"$type $name" for (isconst, type, name)
+                   in zip(arg_consts, arg_ctypes, arg_names)], ", ")
+
+    c_args = join([ifelse(isconst(arg), "const ", "") *
+                   "$(jltoctype[jltype(arg)]) $(name(arg))" for arg in args], ", ")
+
+    "$creturnT $(arbfname(af))($c_args)"
+end
+
+function jlcode(af::Arbfunction, jl_fname=jlfname(af))
```

Comment:
it would be best to keep the external control on what is the function named on julia side;
as is `jl_fname` has no effect on `jlsignature` this control is lost